### ISSUE TITLE
Validation fails when default is set as object property

### ISSUE THREAD

**Issue by luzeno** [stoplightio/spectral#2781]
**Describe the bug**
The oas validation fails when the `default` property is set as an object property, but it doesn't cover all the `required` values:

`38:15    error  oas3-valid-schema-example  "default" property must have required property "count"         components.schemas.Foo.default`.

**To Reproduce**

1. Given this OpenAPI document and Spectral configuration

**api.yaml**
```yaml
openapi: 3.0.3
info:
  title: Sample API
  version: 0.1.0

paths:
  /test:
    post:
      requestBody:
        description: Optional description in *Markdown*
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Foo'
      responses:
        "200": # status code
          description: Sample response
          content:
            application/json:
              schema:
                type: array
                items:
                  type: string

components:
  schemas:
    Foo:
      type: object
      properties:
        name:
          type: string
        count:
          type: integer
      required:
        - name
        - count
      default: # <--- Failing because 'default' doesn't contain the 'count' property that is required
        name: foo
```

**.spectral.yaml**
```yaml
extends: ["spectral:oas"]
```

2. Run this command: `spectral lint api.yaml`
3. Validation fails with the following error:

`38:15    error  oas3-valid-schema-example  "default" property must have required property "count"         components.schemas.Foo.default`

**Expected behavior**
The validation should pass since it is compliant with the [json schema validation](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-6.2) and the [oas specification](https://spec.openapis.org/oas/v3.0.0#properties).

**Environment:**
 - Library version: 6.14.2

**Additional context**
The validation works fine for the given OpenAPI document using the swagger editor online.

**Comment by frankkilcommins** [comment]
Hi @luzeno, 

From a JSON Schema perspective, there's no validation instruction or restriction for the `default` metadata keyword. It is recommended that the _value_ specified does validate against the local schema however.

From https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-6.2:
>    There are no restrictions placed on the value of this keyword.
>
>  This keyword can be used to supply a default JSON value associated
>  with a particular schema.  It is RECOMMENDED that a default value be
>  valid against the associated schema.
>
>  This keyword MAY be used in root schemas, and in any subschemas.

OAS 3.0.* is a subset/superset of JSON Schema Draft 5 and has the following instruction for the `default` keyword:
> The default value represents what would be assumed by the consumer of the input as the value of the schema if one is not provided. Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, if type is "string", then default can be "foo" but cannot be 1.

Let's assume we have the following JSON Schema:
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "properties": {
      "name": {
        "type": "string"
      },
        "count": {
          "type": "integer"
        }
    },
      "required": [
        "name",
        "count"
    ]
 }
```

And we have the following JSON instance:
```
{
   	"name": "foo"
}
```

Validation would return an error along the lines of:
```
Message: Required properties are missing from object: count.
Schema path: #/required
```

I'm not sure what you are trying to achieve by having a default not matching the local schema, as that would likely be confusing for any consumer.

Spectral has certain opinions on what's good practice on top of the various specifications. Hopefully this explains why that's being flagged as an error.
